controllers/state_of_the_world.go

```diff
@@ -392,6 +396,46 @@ func (b *BootOptionsBuilder) getAuthorinoOperatorOptions() []controller.Controll
 	return opts
 }
 
+func (b *BootOptionsBuilder) getObservabilityOptions() []controller.ControllerOption {
+	var opts []controller.ControllerOption
+	var err error
+
+	b.isPrometheusOperatorInstalled, err = utils.IsCRDInstalled(b.manager.GetRESTMapper(), monitoringv1.SchemeGroupVersion.Group, monitoringv1.ServiceMonitorsKind, monitoringv1.SchemeGroupVersion.Version)
+	if err != nil || !b.isPrometheusOperatorInstalled {
+		b.logger.Info("prometheus operator is not installed (ServiceMonitor CRD not found), skipping related watches and reconcilers", "err", err)
+		return opts
+	}
```

Comment:
We also report missing dependency on the affected CRs. Since the field for enabling/disabling obsersability is at the `Kuadrant` CR, do we want to report if the prometheus dependency is missing on the `Kuadrant` CR status also? 🤔 

https://github.com/Kuadrant/kuadrant-operator/blob/5f5dce36389a3f54b3480b599fe2a7aac85927f2/controllers/kuadrant_status_updater.go#L155-L181

Reply:
I am not sure if that is what we want to do all the time. As this is a feature that is enabled in the kuadrant CR I think updating the status to say dependencies are missing would only be required if the feature is enabled. 

I think the status should be addressed but it could be a follow up from this. This is introducing the configuration in the kuadrant CR which I would like to see main so more people could have a look.

Reply:
Yes, I agree that it should be reported only when the feature is enabled and can be done as a follow up